proof_of_concept/definitions/asset_id.py

```diff
@@ -0,0 +1,82 @@
+"""Identifier for assets."""
+from typing import Any, cast, Type
+
+
+class AssetId(str):
+    """An ID of an asset."""
+    def __new__(cls: Type['AssetId'], seq: Any) -> 'AssetId':
+        """Create an AssetId.
```

Comment:
It would be nice to add an example of a primary AssetId and a secondary AssetId, so it's easier to make sense of the code in here.